### ISSUE TITLE
.NET 8 AOT Mac OSX x64 segmentation fault

### ISSUE THREAD

**Issue by chipcode-nl** [dotnet/runtime#92894]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I made the most basic HelloAot, it runs without problem on osx-arm64. But on osx-x64 the program crashes with a segmentation fault.

https://github.com/chipcode-nl/HelloAot

### Expected Behavior

No crash :-)

### Steps To Reproduce

publish the project
run the executable
-> segmentation fault

macOS Ventura 13.5.2 

### Exceptions (if any)

No

### .NET Version

8.0.100-rc.1.23463.5

### Anything else?

.NET SDK:
 Version:   8.0.100-rc.1.23463.5
 Commit:    e7f4de8816

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.5
 OS Platform: Darwin
 RID:         osx-x64
 Base Path:   /usr/local/share/dotnet/sdk/8.0.100-rc.1.23463.5/

.NET workloads installed:
There are no installed workloads to display.

Host:
  Version:      8.0.0-rc.1.23419.4
  Architecture: x64
  Commit:       92959931a3
  RID:          osx-x64

.NET SDKs installed:
  7.0.400 [/usr/local/share/dotnet/sdk]
  8.0.100-rc.1.23463.5 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-rc.1.23421.29 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-rc.1.23419.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by agocke** [comment]
Could this be XCode 15? Did any of you upgrade to the latest XCode recently?

**Comment by agocke** [comment]
Closing as dup of https://github.com/dotnet/runtime/issues/92491 for now. If we find evidence this is not the same bug we can re-open.

**Comment by slang25** [comment]
> TL;DR: The description matches the issues with Xcode 15 linker. We fixed it but there is no official release yet.
> 
> The new linker creates garbage unwinding information, which causes null pointer dereference in `UnixNativeCodeManager`.
> 
> You may be able to workaround it with
> 
> ```
> <ItemGroup>
>   <LinkerArg Include="-Wl,-ld_classic" />
> </ItemGroup>
> ```
> 
> in .csproj (typing on a phone, so there may be typos).

That fixes it 🙂 Thanks @filipnavara

**Comment by chipcode-nl** [comment]
> TL;DR: The description matches the issues with Xcode 15 linker. We fixed it but there is no official release yet.
> 
> The new linker creates garbage unwinding information, which causes null pointer dereference in `UnixNativeCodeManager`.
> 
> You may be able to workaround it with
> 
> ```
> <ItemGroup>
>   <LinkerArg Include="-Wl,-ld_classic" />
> </ItemGroup>
> ```
> 
> in .csproj (typing on a phone, so there may be typos).

This fix  works for me.  Thanks
